### PR TITLE
[Core] Deprecate duplicated Initialize method in find intersected geometrical objects

### PR DESCRIPTION
**Pull request by marcnunezc**
**Description**
The find_intersectred_geometrical_objects_process has two identical intialize methods:

https://github.com/KratosMultiphysics/Kratos/blob/7adf1e3266df392298a517af77f96836e854affd/kratos/processes/find_intersected_geometrical_objects_process.cpp#L82-L85

https://github.com/KratosMultiphysics/Kratos/blob/7adf1e3266df392298a517af77f96836e854affd/kratos/processes/find_intersected_geometrical_objects_process.cpp#L213-L216

This replaces all usages in the core of the Initialize() version and deprecates it. 

**Changelog**
- Replace Initiliaze usages with ExecuteInitialize() in the core
- Deprecate Initialize() method

### CODE REVIEW

**Review by rubenzorrilla — COMMENTED** [review]
I'd add a `KRATOS_WARNING` in the implementation of the `Initialize` so the user gets a warning not only at compile time but at runtime.

**Comment by philbucher** [comment]
is this exposed to python?
If yes then could you please disable the warning in the python interface?

=> e.g. like [here](https://github.com/KratosMultiphysics/Kratos/blob/master/kratos/python/add_processes_to_python.cpp#L150-L152)

**Comment by marcnunezc** [comment]
> I'd add a `KRATOS_WARNING` in the implementation of the `Initialize` so the user gets a warning not only at compile time but at runtime.

Done!

> is this exposed to python?
> If yes then could you please disable the warning in the python interface?
> 
> => e.g. like [here](https://github.com/KratosMultiphysics/Kratos/blob/master/kratos/python/add_processes_to_python.cpp#L150-L152)

This process is not exposed to python

File: kratos/processes/find_intersected_geometrical_objects_process.cpp [hunk 1]
```diff
@@ -81,6 +81,7 @@ FindIntersectedGeometricalObjectsProcess::FindIntersectedGeometricalObjectsProce
 
 void FindIntersectedGeometricalObjectsProcess::Initialize()
 {
+    KRATOS_WARNING("Using FindIntersectedGeometricalObjectsProcess.Initialize() method is deprecated. Please use the ExecuteInitialize() method instead.");
     if (mrModelPartIntersected.NumberOfNodes() > 0) {
         GenerateOctree();
     }
```
Comment (philbucher) [inline kratos/processes/find_intersected_geometrical_objects_process.cpp]:
```suggestion
    this->ExecuteInitialize();
```

Comment (marcnunezc) [reply kratos/processes/find_intersected_geometrical_objects_process.cpp]: Makes sense

**Review by rubenzorrilla — APPROVED** [review]
Fine if CI passes.

**Pull request merged**